scripts/eqtl_hail_batch/generate_eqtl_spearman.py

```diff
@@ -0,0 +1,458 @@
+    geneloc_df = geneloc_df[geneloc_df.gene_name.isin(genes_to_test)]
+
+    return len(geneloc_df.index)
+
+
+def get_log_expression(expression_df):
+
+    """get logged expression values"""
+
+    expression_df = filter_lowly_expressed_genes(expression_df)
+    sample_ids = expression_df.iloc[:, 0]
+    to_log = expression_df.iloc[:, 1:].columns
+    log_expression_df = expression_df[to_log].applymap(lambda x: np.log(x + 1))
+    # subset genes to 10 test genes
+    # remove this when running on entire dataset
+    genes_to_test = ['ERAP2', 'LAMB1', 'POLR2J2', 'GBP7', 'GSTM3', 'CCDC163P', 'CTLA4', 'AIF1', 'BRCA1', 'IL7', 'HDHD5']
+    log_expression_df = log_expression_df[log_expression_df.columns.intersection(genes_to_test)]
+    log_expression_df.insert(loc=0, column='sampleid', value=sample_ids)
+    # remove sc outlier samples
+    log_expression_df = remove_sc_outliers(log_expression_df)
+
+    return log_expression_df
+
+
+def calculate_log_cpm(expression_df, output_prefix):
+    """Calculate log cpm for each cell type and chromosome"""
+
+    expression_df = filter_lowly_expressed_genes(expression_df)
+    sample_ids = expression_df.iloc[:, 0]
+    # remove sampleid column and get log expression
+    # this can only be done on integers
+    expression_df = expression_df.iloc[:, 1:]
+    cpm_df = expression_df.apply(lambda x: (x / sum(x)) * 1000000, axis=0)
+    log_cpm = np.log(cpm_df + 1)
+    # add sampleids back in
+    log_cpm = log_cpm.assign(sampleid=list(sample_ids))
+    log_cpm_path = AnyPath(output_prefix) / 'log_cpm.tsv'
+    with log_cpm_path.open('w') as fp:
+        log_cpm.to_csv(fp, index=False)
+
+
+def prepare_genotype_info(keys_path, expression_path):
+
+    init_batch()
+    filtered_mt_path = output_path('genotype_table.mt', 'tmp')
+    if not hl.hadoop_exists(filtered_mt_path):
+        expression_df = pd.read_csv(AnyPath(expression_path), sep='\t')
+        log_expression_df = get_log_expression(expression_df)
+        mt = hl.read_matrix_table(TOB_WGS)
+        mt = hl.experimental.densify(mt)
+        # filter to biallelic loci only
+        mt = mt.filter_rows(hl.len(mt.alleles) == 2)
+        # filter out variants that didn't pass the VQSR filter
+        mt = mt.filter_rows(hl.len(hl.or_else(mt.filters, hl.empty_set(hl.tstr))) == 0)
+        # VQSR does not filter out low quality genotypes. Filter these out
+        mt = mt.filter_entries(mt.GQ <= 20, keep=False)
+        # filter out samples with a genotype call rate > 0.8 (as in the gnomAD supplementary paper)
+        n_samples = mt.count_cols()
+        call_rate = 0.8
+        mt = mt.filter_rows(
+            hl.agg.sum(hl.is_missing(mt.GT)) > (n_samples * call_rate), keep=False
+        )
+        # filter out variants with MAF < 0.01
+        ht = hl.read_table(FREQ_TABLE)
+        mt = mt.annotate_rows(freq=ht[mt.row_key].freq)
+        mt = mt.filter_rows(mt.freq.AF[1] > 0.01)
```

Comment:
Should this perhaps be >= 0.01 - here we keep anything with a MAF > 0.01, and dropping everything MAF <= 0.01?

Reply:
as below